assets/src/edit-story/components/library/panes/media/mediaUtils.js

```diff
@@ -85,6 +85,140 @@ export const getResourceFromMediaPicker = (mediaPickerEl) => {
   };
 };
 
+/**
+ * Generates a virtual image to pick metadata resource
+ *
+ * @param {Object} image File object
+ * @return {Promise} Resource object
+ */
+
+const getImageMetadata = (image) => {
+  const reader = new window.FileReader();
+
+  return new Promise((resolve, reject) => {
+    reader.onload = function(e) {
+      const src = e.target.result;
+      const randomId = Math.floor(Date.now() / 1000);
+
+      const img = new window.Image();
+
+      img.onload = function() {
+        const width = img.width;
+        const height = img.height;
+
+        resolve({
+          type: 'image',
+          src,
+          width,
+          height,
+          mimeType: image.type,
+          posterId: randomId,
+          poster: '',
+          videoId: undefined,
+          local: true,
+          lengthFormatted: undefined,
+        });
+      };
+
+      img.src = src;
+    };
+
+    reader.onerror = reject;
+
+    reader.readAsDataURL(image);
```

Comment:
Does this sink the stream? I.e. can we use the same stream to both upload and read it here?